src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -349,6 +349,14 @@ protected override void OnHostStarted()
             _secretManager.PurgeOldFiles(Instance.ScriptConfig.RootScriptPath, Instance.TraceWriter);
         }
 
+        protected override void OnTimeout()
+        {
+            base.OnTimeout();
+            Stop();
```

Comment:
@mathewc this is something we looked at a while back and is not needed on the version of the framework we use. With the more recently introduced APIs, `QueueBackgroundWorkItem` will automatically handle the notification and signal the token that is provided to the delegate.